x-pack/plugins/embeddable_enhanced/public/actions/panel_notifications_action.test.ts

```diff
@@ -5,81 +5,113 @@
  * 2.0.
  */
 
-import { PanelNotificationsAction } from './panel_notifications_action';
-import { EnhancedEmbeddableContext } from '../types';
 import { ViewMode } from '@kbn/embeddable-plugin/public';
+import { EmbeddableApiContext, PublishesWritableViewMode } from '@kbn/presentation-publishing';
+import { DynamicActionManager } from '@kbn/ui-actions-enhanced-plugin/public/dynamic_actions';
+import { BehaviorSubject } from 'rxjs';
+import { HasDynamicActions } from '../embeddables/interfaces/has_dynamic_actions';
+import { DynamicActionsSerializedState } from '../plugin';
+import {
+  PanelNotificationsAction,
+  PanelNotificationsActionApi,
+} from './panel_notifications_action';
+
+const createContext = (events: any[] = []): EmbeddableApiContext => {
+  const dynamicActionsState$ = new BehaviorSubject<DynamicActionsSerializedState['enhancements']>({
+    dynamicActions: { events },
+  });
+  const viewMode$ = new BehaviorSubject<ViewMode>(ViewMode.VIEW);
 
-const createContext = (events: unknown[] = [], isEditMode = false): EnhancedEmbeddableContext =>
-  ({
+  return {
     embeddable: {
-      getInput: () =>
-        ({
-          viewMode: isEditMode ? ViewMode.EDIT : ViewMode.VIEW,
-        } as unknown),
       enhancements: {
-        dynamicActions: {
-          state: {
-            get: () =>
-              ({
-                events,
-              } as unknown),
-          },
-        },
+        dynamicActions: {} as unknown as DynamicActionManager,
+      },
+      setDynamicActions: (value: DynamicActionsSerializedState['enhancements']) => {
+        dynamicActionsState$.next(value);
       },
-    },
-  } as EnhancedEmbeddableContext);
+      dynamicActionsState$,
+      viewMode: viewMode$,
+      setViewMode: (value: ViewMode) => viewMode$.next(value),
+    } as PanelNotificationsActionApi & PublishesWritableViewMode,
+  } as EmbeddableApiContext;
+};
 
 describe('PanelNotificationsAction', () => {
   describe('getDisplayName', () => {
     test('returns "0" if embeddable has no events', async () => {
       const context = createContext();
       const action = new PanelNotificationsAction();
 
-      const name = await action.getDisplayName(context);
+      const name = action.getDisplayName(context);
       expect(name).toBe('0');
     });
 
     test('returns "2" if embeddable has two events', async () => {
       const context = createContext([{}, {}]);
       const action = new PanelNotificationsAction();
 
-      const name = await action.getDisplayName(context);
+      const name = action.getDisplayName(context);
       expect(name).toBe('2');
     });
+
+    test('updates display name when dynamic actions is updated', async () => {
+      const context = createContext([{}, {}]);
+      const action = new PanelNotificationsAction();
+
+      (context.embeddable as HasDynamicActions).setDynamicActions({
+        dynamicActions: { events: [{}, {}, {}] as any[] },
```

Comment:
type `as  unknown as SerializedEvent[]`

Reply:
Done in 343e464a979251cae24177f87eb348c52ab673b0